ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -141,34 +145,22 @@ void *libX11_Variadic_u64(uint64_t a_0, size_t count, unsigned long *list, void
       # Store how much data we added to the stack in our callee saved register we stole
       mov x28, x10
 
-      # Subtract one member due to nullptr ender
-      sub x10, x1, 1
-
-      # x11 - stack offset
+      # x11 - stack offset - for stack Arguments
       mov x11, sp
 
-      # x12 - load offset
+      # x12 - load offset into input array (skipping the first 7 arguments)
       add x12, x9, (7 * 8)
 
-      cmp x10, 1
-      b.eq .single%=
-      b.lt .no_single%=
+      # Compute the number of elements excluding the terminating nullptr
+      subs x10, x1, 1
 
-      .load_pair%=:
-      ldp x1, x2, [x12], 16
-      stp x1, x2, [x11], 16
-      sub x10, x10, 8
-      cmp x10, 1
-      b.gt .load_pair%=
-      b.lt .no_single%=
-
-      # One variable at most
       .single%=:
-      ldr x1, [x12]
-      stp x1, xzr, [x11]
-      b .top_reg_args%=
+      ldr x1, [x12], 8
+      str x1, [x11], 8
+      sub x10, x10, 1
+      cmp x10, 1
```

Comment:
This seems to copy one element if `count` is 7 now, but IIUC it should only write the nullptr in that case. Could use `ldr.ge`/`str.ge` instead, I guess?

Reply:
ldr.ge, str.ge aren't real.